ArmPkg/Drivers/CpuDxe/AArch64/Mmu.c

```diff
@@ -168,16 +167,11 @@ GetNextEntryAttribute (
   for (Index = 0; Index < EntryCount; Index++) {
     Entry = TableAddress[Index];
 
-    // MU_CHANGE [BEGIN]: Convert integers safely
-    Status = SafeUint64ToUint32 (Entry, &EntryType);
-    if (EFI_ERROR (Status)) {
-      DEBUG ((DEBUG_ERROR, "[%a] - Table address entry exceeds 32-bit.\n", __func__));
-      return 0;
-    }
-
-    EntryAttribute = EntryType & TT_ATTRIBUTES_MASK;  // MU_CHANGE: Return all attributes from page table
-    EntryType     &= TT_TYPE_MASK;
-    // MU_CHANGE [END]: Convert integers safely
+    // MU_CHANGE [BEGIN]: Add UINT32 cast
+    EntryType      = (UINT32)(Entry & TT_TYPE_MASK);

```

Comment:
I might have missed something, but what is stopping us from changing `EntryType` to UINT32.

Reply:
`EntryType` is a `UINT32` at the moment. My understanding is that this is just the lower 2 bits being masked off from the descriptor.

Where `0b11` is here:
![image](https://github.com/microsoft/mu_silicon_arm_tiano/assets/21320094/4d9a50a5-c8d9-4223-ba69-5c0eaa7e576e)

```c
#define TT_TYPE_MASK                0x3
```

It is only compared to these values in the function:

```c
#define TT_TYPE_TABLE_ENTRY         0x3
#define TT_TYPE_BLOCK_ENTRY         0x1
#define TT_TYPE_BLOCK_ENTRY_LEVEL3  0x3
```

So, I didn't see a need to change its type.

Reply:
Hmm, I think Arm plays the game of casting with too much liberty. My thought is that we should not do casting unless we have to (UINTN to specific size), and it seems that this variable is better to be updated to avoid casting completely.

Reply:
> Hmm, I think Arm plays the game of casting with too much liberty. 

Definitely, it's a problem in a lot of the Arm code. That's essentially what this PR does. It makes `EntryAttribute` a `UINT64` to avoid a cast.

For `EntryType` specifically, it could also be a `UIN64` given its usage, but it's not being passed to another function that expects and uses 64 bits like `EntryAttribute`.

I'm happy to make `EntryType` a `UINT64` though given its limited scope to this function.

Reply:
Done in PR commit cb0253ddbd